lib/msf/core/handler/reverse_tcp_comm.rb

```diff
@@ -0,0 +1,41 @@
+# -*- coding: binary -*-
+require 'rex/socket'
+
+module Msf
+module Handler
+
+###
+#
+# This module implements the reverse Rex::Socket::Comm handlng.
+#
+###
+module ReverseTcpComm
+
+  def initialize(info = {})
+    super
+
+    register_advanced_options(
+      [
+        OptString.new('ReverseListenerComm', [ false, 'The specific communication channel to use for this listener']),
+      ], Msf::Handler::ReverseTcpComm)
+  end
+
+  def select_comm
+    rl_comm = datastore['ReverseListenerComm'].to_s
+    case rl_comm
+    when 'local'
+      comm = ::Rex::Socket::Comm::Local
+    when /\A[0-9]+\Z/
+      comm = framework.sessions[rl_comm.to_i]
+      raise(RuntimeError, "Reverse Listener Comm (Session #{rl_comm}) does not exist") unless comm
+      raise(RuntimeError, "Reverse Listener Comm (Session #{rl_comm}) does not implement Rex::Socket::Comm") unless comm.is_a? ::Rex::Socket::Comm
+    else
```

Comment:
It seems like if an invalid value is specified for ```ReverseListenerComm``` such as "fake", that this should fail like it does above when the numerical value is incorrect.